cpp/src/strings/json/json_to_array.cu

```diff
@@ -0,0 +1,563 @@
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/get_value.cuh>
+#include <cudf/detail/utilities/cuda.cuh>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/scalar/scalar_factories.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/strings/detail/utilities.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/error.hpp>
+#include <cudf/structs/structs_column_view.hpp>
+
+#include <rmm/exec_policy.hpp>
+
+namespace cudf {
+
+class parser {
+  protected:
+  CUDA_HOST_DEVICE_CALLABLE parser() : input(nullptr), input_len(0), pos(nullptr) {}
+  CUDA_HOST_DEVICE_CALLABLE parser(const char* _input, int64_t _input_len)
```

Comment:
Why are all of these `host/device` functions? Are they used in host and device code?

Reply:
See my comment below about get_json_object().  This code came from an early version of that.  I had these all marked as host/device for debugging purposes (being able to invoke it from CPU/GPU).